pkg/storage/trb_request_feedback.go

```diff
@@ -0,0 +1,156 @@
+			zap.String("user", feedback.CreatedBy),
+		)
+		return nil, err
+	}
+
+	// If the feedback requests edits, update the form status to "in progress"
+	if formToUpdate != nil {
+		formStmt, formErr := tx.PrepareNamed(`
+			UPDATE trb_request_forms
+			SET
+				status = :status,
+				modified_by = :modified_by,
+				modified_at = CURRENT_TIMESTAMP
+			WHERE trb_request_id = :trb_request_id
+			RETURNING *;
+		`)
+
+		updatedForm := models.TRBRequestForm{}
+		formErr = formStmt.Get(&updatedForm, formToUpdate)
+
+		if formErr != nil {
+			appcontext.ZLogger(ctx).Error(
+				fmt.Sprintf("Failed to update TRB request form when creating TRB request feedback, with error %s", err),
+				zap.String("user", formToUpdate.CreatedBy),
+			)
+			return nil, formErr
+		}
+	}
+	err = tx.Commit()
+
+	if err != nil {
+		appcontext.ZLogger(ctx).Error("Failed to create TRB request with error %s", zap.Error(err))
+		return nil, err
+	}
+
+	return &createdFeedback, nil
+}
+
+// GetTRBRequestFeedbackByTRBRequestID queries the DB for all the TRB request feedback records
+// matching the given TRB request ID
+func (s *Store) GetTRBRequestFeedbackByTRBRequestID(ctx context.Context, trbRequestID uuid.UUID) ([]*models.TRBRequestFeedback, error) {
+	results := []*models.TRBRequestFeedback{}
+
+	err := s.db.Select(&results, `SELECT * FROM trb_request_feedback WHERE trb_request_id = $1`, trbRequestID)
+
+	if err != nil && !errors.Is(err, sql.ErrNoRows) {
+		appcontext.ZLogger(ctx).Error("Failed to fetch TRB request feedback", zap.Error(err), zap.String("id", trbRequestID.String()))
+		return nil, &apperrors.QueryError{
+			Err:       err,
+			Model:     models.TRBRequestFeedback{},
+			Operation: apperrors.QueryFetch,
+		}
+	}
+	return results, nil
+}
+
+// GetNewestTRBRequestFeedbackByTRBRequestID queries the DB for all the TRB request feedback records
+// matching the given TRB request ID
+func (s *Store) GetNewestTRBRequestFeedbackByTRBRequestID(ctx context.Context, trbRequestID uuid.UUID) (*models.TRBRequestFeedback, error) {
+	feedback := models.TRBRequestFeedback{}
+	stmt, err := s.db.PrepareNamed(`
+		SELECT *
+		FROM trb_request_feedback
+		WHERE trb_request_id = :trb_request_id
+		ORDER BY created_at DESC LIMIT 1
+	`)
+
+	if err != nil {
+		return nil, err
+	}
+	arg := map[string]interface{}{"trb_request_id": trbRequestID}
+	err = stmt.Get(&feedback, arg)
+
+	hasFeedback := true
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			hasFeedback = false
+		} else {
+			appcontext.ZLogger(ctx).Error(
+				"Failed to fetch latest TRB request feedback",
+				zap.Error(err),
+				zap.String("trbRequestID", trbRequestID.String()),
+			)
+			return nil, &apperrors.QueryError{
+				Err:       err,
+				Model:     feedback,
+				Operation: apperrors.QueryFetch,
+			}
+		}
+	}
+
+	// If no rows, return nil
+	var feedbackPtr *models.TRBRequestFeedback
+	if hasFeedback {
+		feedbackPtr = &feedback
+	} else {
+		feedbackPtr = nil
+	}
+	return feedbackPtr, nil
```

Comment:
Minor nitpick: I don't think you need to explicitly create `feedbackPtr` here, unless I'm missing something; this should work, and is a bit more concise:
```go
if hasFeedback {
	return &feedback, nil
} else {
	return nil, nil
}
```

Reply:
I prefer to just have one return statement

Reply:
Gotcha, no problem.